Automattic/wp-calypso
# Tour Kit: Update spotlight effect, other fixes/cleanup

#### Changes proposed in this Pull Request

* Updates spotlight effect 
    * now being reference element aware (wraps the entire reference by default)
    * customizable (can be overwritten with custom CSS props e.g. in Welcome Tour)
* Updates Storybook examples
    * capture placement on all sides
    * resizable container for fiddling with scrolling etc.
* Other fixes 
    * issue with the arrow tip indicator 
    * issue with Popper misplacing the step on mount/ initial render

#### Media

Default spotlight:

![Kapture 2021-12-06 at 15 30 00](https://user-images.githubusercontent.com/1705499/144854665-3072a4d8-1259-4598-9fcc-f1f25b94fe29.gif)

Customized for Welcome Tour (fixed-size spotlight, some border radius):

<img width="512" alt="Screenshot 2021-12-06 at 3 33 28 PM" src="https://user-images.githubusercontent.com/1705499/144855167-3c644d30-03cc-4e52-84cb-f98fa0196b65.png">

#### Testing instructions



* Checkout branch, then `yarn dev --sync` from `apps/editing-toolkit` to sync the welcome tour to the sandbox
* `yarn run tour-kit:storybook:start` in another tab for Storybook demos
* Welcome Tour works as expected with/out `?welcome-tour-next`
* Stories work as expected



Closes #58642, #58748
Related to #58644
